ocaml/xapi/context.ml

```diff
@@ -227,8 +227,64 @@ let parent_of_origin (origin : origin) span_name =
   | _ ->
       None
 
+let attribute_helper_fn f v = Option.fold ~none:[] ~some:f v
+
+let addr_port_of_sock s =
+  match s with
+  | Unix.ADDR_UNIX "" ->
+      (None, None)
+  | Unix.ADDR_UNIX soket_name ->
+      (Some soket_name, None)
+  | Unix.ADDR_INET (addr, port) ->
+      (Some (Unix.string_of_inet_addr addr), Some (string_of_int port))
+
+let attr_of_fd s =
+  let peer_addr = s |> Unix.getpeername |> addr_port_of_sock in
+  let local_addr = s |> Unix.getsockname |> addr_port_of_sock in
+  [
+    attribute_helper_fn
+      (fun addr -> [("network.local.address", addr)])
+      (fst local_addr)
+  ; attribute_helper_fn
+      (fun port -> [("network.local.port", port)])
+      (snd local_addr)
+  ; attribute_helper_fn
+      (fun addr -> [("network.peer.address", addr)])
+      (fst peer_addr)
+  ; attribute_helper_fn
+      (fun peer_addr -> [("network.peer.port", peer_addr)])
+      (snd peer_addr)
+  ]
+  |> List.concat
+
+let attr_of_req (req : Http.Request.t) =
+  [
+    [
+      ("xs.xapi.task.origin", "http")
+    ; ("http.request.header.method", Http.string_of_method_t req.m)
+    ]
+  ; attribute_helper_fn
+      (fun user_agent -> [("http.request.header.user-agent", user_agent)])
+      req.user_agent
+  ; attribute_helper_fn
+      (fun content_type -> [("http.request.header.content-type", content_type)])
+      req.content_type
+  ; attribute_helper_fn
+      (fun content_length ->
+        [("http.request.body.size", Printf.sprintf "%Li" content_length)]
+      )
+      req.content_length
+  ; List.map
+      (fun (h, v) ->
+        ( h |> String.lowercase_ascii |> Printf.sprintf "http.request.header.%s"
+        , v
+        )
+      )
+      req.additional_headers
+  ]
+  |> List.concat
+
 let make_attributes ?task_name ?task_id ?task_uuid ?session_id ?origin () =
-  let attribute_helper_fn f v = Option.fold ~none:[] ~some:f v in
   [
     attribute_helper_fn
       (fun task_name -> [("xs.xapi.task.name", task_name)])
```

Comment:
xs.xapi.session.trackid somewhere here?

Reply:
A few lines below. Line 295 on the changes. It was already added in another PR.